### PR TITLE
test(looker): add `retail_demo` project

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
Copy a subset of https://github.com/looker-open-source/retail_demo for unit testing. Included the license as well.

## How I Tested These Changes
See downstream PRs.

### CODE REVIEW

**Comment by rexledesma** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21293?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#21294** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21294?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21293** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21293?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#21284** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21284?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rexledesma and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-looker/dagster_looker_tests/looker_projects/retail_demo/dashboards/.gitkeep
Comment (sryza) [inline python_modules/libraries/dagster-looker/dagster_looker_tests/looker_projects/retail_demo/dashboards/.gitkeep:1]: why is this needed?

**Review by sryza — COMMENTED** [review]
I am not an expert on this stuff, but I think we might need to include the license from the repo that we're copying this from.

File: python_modules/libraries/dagster-looker/dagster_looker_tests/looker_projects/retail_demo/dashboards/.gitkeep
Comment (rexledesma) [reply python_modules/libraries/dagster-looker/dagster_looker_tests/looker_projects/retail_demo/dashboards/.gitkeep:1]: removed. it was just copied from the repository.

**Review by benpankow — APPROVED** [review]
Seems like there are some changes to the `dagster_looker` boilerplate showing up here too, maybe needs a rebase?

**Comment by rexledesma** [comment]
### Merge activity

* **May 2, 9:48 AM PDT**: @rexledesma merged this pull request with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/21293).
